### PR TITLE
CDD-1843: Lookup authenticated eori from api-subscription-fields

### PR DESCRIPTION
**Pull request by rashadmughal**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/inventorylinking/imports/model/ApiSubscriptionFieldsResponse.scala [hunk 1]
```diff
@@ -20,8 +20,14 @@ import java.util.UUID
 
 import play.api.libs.json.{Json, OFormat}
 
-case class ApiSubscriptionFieldsResponse(fieldsId: UUID)
+case class ApiSubscriptionFieldsResponse(fieldsId: UUID, fields: ApiSubscriptionFieldsResponseFields)
+
+case class ApiSubscriptionFieldsResponseFields(authenticatedEori: Option[String])
 
 object ApiSubscriptionFieldsResponse {
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/inventorylinking/imports/model/ApiSubscriptionFieldsResponse.scala]: I think `Oformat` can just `Format`

File: app/uk/gov/hmrc/customs/inventorylinking/imports/services/MessageSender.scala [hunk 2]
```diff
@@ -53,18 +53,23 @@ class MessageSender @Inject()(apiSubscriptionFieldsConnector: ApiSubscriptionFie
...
     }
   }
 
   private def futureApiSubFieldsId[A](c: ClientId)
-                                     (implicit vpr: ValidatedPayloadRequest[A], hc: HeaderCarrier): Future[Either[Result, FieldsId]] = {
+                                     (implicit vpr: ValidatedPayloadRequest[A], hc: HeaderCarrier): Future[Either[Result, ApiSubscriptionFieldsResponse]] = {
     (apiSubscriptionFieldsConnector.getSubscriptionFields(ApiSubscriptionKey(c, apiContextEncoded, VersionOne)) map {
       response: ApiSubscriptionFieldsResponse =>
-        Right(FieldsId(response.fieldsId.toString))
+        logger.debug(s"Got a response from api subscription fields $response")
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/inventorylinking/imports/services/MessageSender.scala]: I suggest re-writing `response.fields.authenticatedEori match` as `response.fields.authenticatedEori.fold(...`

File: app/uk/gov/hmrc/customs/inventorylinking/imports/xml/PayloadDecorator.scala [hunk 3]
```diff
@@ -36,13 +35,19 @@ class PayloadDecorator {
...
       <n1:requestCommon>
-        <n1:clientID>{ clientId.value }</n1:clientID>
+        <n1:clientID>{ apiSubscriptionFieldsResponse.fieldsId.toString }</n1:clientID>
         <n1:conversationID>{ vpr.conversationId.toString}</n1:conversationID>
         <n1:correlationID>{ correlationId.toString }</n1:correlationID>
         <n1:badgeIdentifier>{ vpr.badgeIdentifier.value }</n1:badgeIdentifier>
         <n1:dateTimeStamp>{ dateTime.toString(ISODateTimeFormat.dateTimeNoMillis) }</n1:dateTimeStamp>
         <n1:submitterID>{ vpr.submitterIdentifier.value }</n1:submitterID>
-        <n1:authenticatedpartyID>{ authenticatedEori.value }</n1:authenticatedpartyID>
+        {
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/inventorylinking/imports/xml/PayloadDecorator.scala]: I suggest rewriting this as a `apiSubscriptionFieldsResponse.fields.authenticatedEori .fold`

File: app/uk/gov/hmrc/customs/inventorylinking/imports/model/ApiSubscriptionFieldsResponse.scala [hunk 4]
```diff
@@ -20,8 +20,14 @@ import java.util.UUID
 
 import play.api.libs.json.{Json, OFormat}
 
-case class ApiSubscriptionFieldsResponse(fieldsId: UUID)
+case class ApiSubscriptionFieldsResponse(fieldsId: UUID, fields: ApiSubscriptionFieldsResponseFields)
+
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/inventorylinking/imports/model/ApiSubscriptionFieldsResponse.scala]: I suggest renaming `authenticatedEori` to `maybeAuthenticatedEori`

Comment (rashadmughal) [reply app/uk/gov/hmrc/customs/inventorylinking/imports/model/ApiSubscriptionFieldsResponse.scala]: the field is called authenticatedEori so can't changed the name

Comment (googley42) [reply app/uk/gov/hmrc/customs/inventorylinking/imports/model/ApiSubscriptionFieldsResponse.scala]: good point, eloquently made

**Pull request merged**